app/controllers/api/v1x2/icons_controller.rb

```diff
@@ -0,0 +1,16 @@
+module Api
+  module V1x2
+    class IconsController < Api::V1x1::IconsController
+
+      def raw_icon
+        image = find_icon(params.permit(:icon_id, :portfolio_item_id, :portfolio_id, :cache_id)).image.decoded_image
```

Comment:
You can refactor `params.permit(:icon_id, :portfolio_item_id, :portfolio_id, :cache_id)` into a method `parse_params` in both v1.1 and v1.2. Then you don't need to override `raw_icon` with almost identical code.

Reply:
I will do that on Wednesday (I am off tomorrow).